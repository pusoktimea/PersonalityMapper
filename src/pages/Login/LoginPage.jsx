import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';

import Label from 'components/Label';
import Input from 'components/Input';
import Button from 'components/Button';
import Icon from 'components/Icon';

import './style.scss';

class Login extends PureComponent {
  static propTypes = {
    history: PropTypes.object,
    onDoLogin: PropTypes.func.isRequired,
    isLoading: PropTypes.bool.isRequired,
    hasLoginError: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string.isRequired
  };

  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: ''
    };
  }

  loginHandler = () => {
    this.props.onDoLogin({...this.state}, () => this.props.history.push('/dashboard'));
  }

  render() {
    const {isLoading,
      hasLoginError,
      errorMessage
    } = this.props;
    const {
      username,
      password
    } = this.state;

    return (
      <div className="activate-login">
        <div className="activate-login_form">
          <h2 className="activate-login_form_title">Sign in to <span className="activate-login_form_title_logo">Activate</span><span className="activate-login_form_title_logo_bold">MS</span></h2>
          <form onSubmit={this.loginHandler}>
            <div className="activate-login_form_input">
              <Label>
                <Input
                  icon="user"
                  iconPosition="left"
                  name="test_input"
                  placeholder="Username"
                  theme="dark"
                  type="text"
                  onChange={(value) => this.setState({username: value})}
                />
              </Label>
            </div>
            <div className="activate-login_form_input">
              <Label>
                <Input
                  icon="lock"
                  iconPosition="left"
                  name="test_input"
                  placeholder="Password"
                  theme="dark"
                  type="password"
                  onChange={(value) => this.setState({password: value})}
                />
              </Label>
              {
                hasLoginError &&
                  <p className="activate-login_form_error-message">{errorMessage}</p>
              }
            </div>
            <div className="activate-login_form_input">
              <Button
                disabled={isLoading || !username || !password}
                theme="primary"
                onClick={this.loginHandler}
                isLoading={isLoading}
              >
                <Icon icon="sign-in" />
                Sign In
              </Button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default Login;
