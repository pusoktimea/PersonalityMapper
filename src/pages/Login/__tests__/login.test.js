import React from 'react';
import renderer from 'react-test-renderer';

import LoginPage from '../LoginPage';

describe('Login Page test', () => {
  it('initial load - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <LoginPage
          onDoLogin={jest.fn()}
          isLoading={false}
          hasLoginError={false}
          errorMessage=""
        />
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('login in progress - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <LoginPage
          onDoLogin={jest.fn()}
          isLoading
          hasLoginError={false}
          errorMessage=""
        />
      ).toJSON()
    ).toMatchSnapshot();
  });

  it('with error - renders and matches snapshot', () => {
    expect(
      renderer.create(
        <LoginPage
          onDoLogin={jest.fn()}
          isLoading={false}
          hasLoginError
          errorMessage="login failed"
        />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
