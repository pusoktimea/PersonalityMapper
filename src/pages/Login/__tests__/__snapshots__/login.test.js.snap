// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page test initial load - renders and matches snapshot 1`] = `
<div
  className="activate-login"
>
  <div
    className="activate-login_form"
  >
    <h2
      className="activate-login_form_title"
    >
      Sign in to 
      <span
        className="activate-login_form_title_logo"
      >
        Activate
      </span>
      <span
        className="activate-login_form_title_logo_bold"
      >
        MS
      </span>
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-user"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-lock"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        className="activate-login_form_input"
      >
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <span
            className="activate-icon fa fa-sign-in"
            id={undefined}
            onClick={undefined}
          />
          Sign In
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login Page test login in progress - renders and matches snapshot 1`] = `
<div
  className="activate-login"
>
  <div
    className="activate-login_form"
  >
    <h2
      className="activate-login_form_title"
    >
      Sign in to 
      <span
        className="activate-login_form_title_logo"
      >
        Activate
      </span>
      <span
        className="activate-login_form_title_logo_bold"
      >
        MS
      </span>
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-user"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-lock"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        className="activate-login_form_input"
      >
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled activate-button--loading"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <img
            className="activate-button_loading-icon"
            src={Object {}}
          />
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`Login Page test with error - renders and matches snapshot 1`] = `
<div
  className="activate-login"
>
  <div
    className="activate-login_form"
  >
    <h2
      className="activate-login_form_title"
    >
      Sign in to 
      <span
        className="activate-login_form_title_logo"
      >
        Activate
      </span>
      <span
        className="activate-login_form_title_logo_bold"
      >
        MS
      </span>
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-user"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
            />
          </div>
        </label>
      </div>
      <div
        className="activate-login_form_input"
      >
        <label
          className="activate-label"
        >
          <div
            className="activate-input activate-input--dark activate-input--with-icon activate-input--icon-left"
          >
            <span
              className="activate-icon fa fa-lock"
              id={undefined}
              onClick={undefined}
            />
            <input
              className="activate-input_input"
              disabled={false}
              name="test_input"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
        </label>
        <p
          className="activate-login_form_error-message"
        >
          login failed
        </p>
      </div>
      <div
        className="activate-login_form_input"
      >
        <button
          className="activate-button activate-button--primary activate-button--size-default activate-button--disabled"
          disabled={true}
          onClick={[Function]}
          to={undefined}
        >
          <span
            className="activate-icon fa fa-sign-in"
            id={undefined}
            onClick={undefined}
          />
          Sign In
        </button>
      </div>
    </form>
  </div>
</div>
`;
